source/extensions/filters/common/expr/custom_cel/custom_cel_vocabulary.h

```diff
@@ -0,0 +1,162 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+#include "envoy/config/typed_config.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/filters/common/expr/context.h"
+
+#include "absl/strings/string_view.h"
+#include "eval/public/activation.h"
+#include "eval/public/cel_function_registry.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+namespace Custom_CEL {
+
+using google::api::expr::runtime::Activation;
+using google::api::expr::runtime::CelFunctionRegistry;
+
+// CustomCELVocabulary
+//
+// Background Information:
+//
+// CelExpression: CelExpressions can contain values, variables and functions.
+// The functions can be (1) lazy functions or (2) static (eagerly evaluated) functions.
+// Lazy functions are evaluated during the request.
+// Static functions are evaluated once during the duration of the application.
```

Comment:
What is the application here? It might be better to also add some text to relate this to the request and extension lifecycle.